Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -43,17 +64,24 @@ RhoFunctor::operator() ( amrex::MultiFab& mf_dst, const int dcomp, const int /*i
     // Dump total rho
     if (m_species_index == -1) {
         auto& mypc = warpx.GetPartContainer();
-        rho = mypc.GetChargeDensity(m_lev, true);
+        for (int lev=m_lev; lev<=warpx.finestLevel(); lev++) {
+            rho_fp[lev] = mypc.GetChargeDensity(lev, true);
+        }
     }
     // Dump rho per species
     else {
         auto& mypc = warpx.GetPartContainer().GetParticleContainer(m_species_index);
-        rho = mypc.GetChargeDensity(m_lev, true);
+        for (int lev=m_lev; lev<=warpx.finestLevel(); lev++) {
+            rho_fp[lev] = mypc.GetChargeDensity(lev, true);
+        }
     }
 
-    // Handle the parallel transfers of guard cells and
-    // apply the filtering if requested.
-    warpx.ApplyFilterandSumBoundaryRho(m_lev, m_lev, *rho, 0, rho->nComp());
+    // Handle parallel transfers of guard cells,
+    // summation between levels, and filtering
+    warpx.SyncRho(rho_fp, rho_cp, rho_buf, m_lev);
```

Comment:
We call `SyncRho` with `bottom_level=m_lev`. This is a small optimization: if we want `rho` at `m_lev`, we only need to sum `rho` from the highest level down to `lev=m_lev` - no need to go all the way down to `lev=0`.